### ISSUE TITLE
VSCode’s renderer helper process using 100% of CPU

### ISSUE THREAD

**Issue by chenbin3625** [microsoft/vscode#148226]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version:1.66.2
- OS Version: Darwin arm64 21.4.0
- 
提交: dfd34e8260c270da74b5c2d86d61aee4b6d56977
日期: 2022-04-11T07:47:29.396Z
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0

<img width="912" alt="image" src="https://user-images.githubusercontent.com/56670467/165425065-a5d20b8d-d916-46ca-88d6-8926b3bf84c2.png">
<img width="1140" alt="image" src="https://user-images.githubusercontent.com/56670467/165425121-24024f43-7bd4-4115-91c7-40af47685146.png">

The vscode Code Helper (Renderer) process consumes a lot of cpu and causes a very high processor temperature, closing vscode and disabling all extensions will not solve the problem.
After a clean reinstall the problem persisted. The way to reproduce is to open vscode and use the search function will appear, and vscode's own process manager does not show this abnormal process, you need to manually go to apple's process manager to end the task.

**Comment by chenbin3625** [comment]
<img width="698" alt="image" src="https://user-images.githubusercontent.com/56670467/165427826-50fe86cb-dec1-46d1-9263-261ff4d49dd3.png">
My Computer.
MacBook Air 2020 M1
16G RAM + 512G ROM
If there's any other information you need, I'd be happy to provide it.

**Comment by Moshikot3** [comment]
Having the same issue on MacBook Pro 2020 M1 16GB RAM

it took 30% of my CPU for no reason while VS Code was closed.

Feel free to ask for information / logs if needed.

**Comment by deepak1556** [comment]
Can you attach a process sample from the activity monitor of the `Code Helper (Renderer)` process that shows high cpu usage.

**Comment by chenbin3625** [comment]
> 您能否从`Code Helper (Renderer)`显示高 cpu 使用率的进程的活动监视器中附加一个进程示例。
This process is not visible in vscode's process manager. And this process will not be closed after closing vscode. There is no way to know which task is the host of this process.

**Comment by chenbin3625** [comment]
> 您可以从`Code Helper (Renderer)`高 cpu 使用率进程的活动中显示附加进程的一个示例。

The problem is also not that it takes up 100% of one core usage, but because this process causes the MacBook to heat up abnormally. Normally, the MacBook's cpu temperature is around 40 degrees Celsius, and once this abnormal process is in place, the MacBook's cpu temperature will soar to 80 degrees Celsius.

**Comment by fanyang-mono** [comment]
I also hit this issue on my MacBook Pro (15-inch, 2018) even after I exit VS Code completely.

**Comment by richard-fairthorne** [comment]
I have the same issue on a brand new M1 Mac with no other software installed after purchase. No source code has ever been opened. No extensions have been added. I had a similar issue on my 2012 MacBook Pro Retina, and purchased the M1 yesterday to make the issue tolerable.

Yesterday I purchased the machine, and installed VSCode. I opened it once to see that it operated, and then closed it. Not long after, I closed the machine.

This morning, I opened the machine. It got warm quickly, and these processes were hogging cores. NB, VSCode is NOT running, and no extensions have been added.

Rogue processes:

<img width="954" alt="Screen Shot 2022-05-12 at 7 20 21 AM" src="https://user-images.githubusercontent.com/1225460/168064354-c96f6c07-e5bf-48df-af21-6f7c72ba7095.png">

Process information:

<img width="703" alt="Screen Shot 2022-05-12 at 7 21 05 AM" src="https://user-images.githubusercontent.com/1225460/168064546-391d6bd8-2433-4f1e-a952-f62631984cb6.png">

Macbook information:

<img width="594" alt="Screen Shot 2022-05-12 at 7 21 52 AM" src="https://user-images.githubusercontent.com/1225460/168064555-0b809cd1-e4a0-4b05-9222-6bf4f0807eca.png">

Here is the VSCode version information:

```
Version: 1.67.1 (Universal)
Commit: da15b6fd3ef856477bf6f4fb29ba1b7af717770d
Date: 2022-05-06T12:37:16.526Z
Electron: 17.4.1
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Darwin arm64 21.3.0
```

Both my new and old MacBooks are always running hot due to these processes, which is more desirable during the winter, but it's May 🤣

**Comment by richard-fairthorne** [comment]
Upon opening a new instance of VSCode, these processes do not appear in process explorer.

<img width="805" alt="Screen Shot 2022-05-12 at 7 34 47 AM" src="https://user-images.githubusercontent.com/1225460/168065799-404ba704-faca-466d-8388-aa3251eddf16.png">

**Comment by RiccardoGrigoletto** [comment]
Same Issue here with M1 Pro. 
Note that, in my case, I installed VS Code 1 day ago and not used it at all, I just opened the program once to verify the installation. 

![Screenshot 2022-06-14 at 08 36 37](https://user-images.githubusercontent.com/18590116/173510006-c2ee7d69-8dd6-4777-95ec-dd03da761653.png)
![Screenshot 2022-06-14 at 08 37 05](https://user-images.githubusercontent.com/18590116/173510012-2a7c514f-9f1f-46c2-aeb1-71dd45f45722.png)
`rigr@Riccardos-MacBook-Pro ~ % ps aux | grep 5009
rigr              5009 397.1  0.1 456362256  10848   ??  R    10:48AM 1050:07.39 /private/var/folders/rx/l1_q8xhj5fl14ktr8m0gtb_40000gn/T/AppTranslocation/6FDA13BD-289C-4EB1-AA92-6A8898E39F87/d/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer) --type=renderer --user-data-dir=/Users/rigr/Library/Application Support/Code --standard-schemes=vscode-webview,vscode-file --secure-schemes=vscode-webview,vscode-file --bypasscsp-schemes --cors-schemes=vscode-webview,vscode-file --fetch-schemes=vscode-webview,vscode-file --service-worker-schemes=vscode-webview --streaming-schemes --app-path=/private/var/folders/rx/l1_q8xhj5fl14ktr8m0gtb_40000gn/T/AppTranslocation/6FDA13BD-289C-4EB1-AA92-6A8898E39F87/d/Visual Studio Code.app/Contents/Resources/app --no-sandbox --no-zygote --node-integration-in-worker --disable-color-correct-rendering --lang=en-GB --num-raster-threads=4 --enable-zero-copy --enable-gpu-memory-buffer-compositor-resources --enable-main-frame-before-activation --renderer-client-id=5 --launch-time-ticks=4271716250 --shared-files --field-trial-handle=1718379636,2816795883407190807,8334812983944564800,131072 --disable-features=PlzServiceWorker,SpareRendererForSitePerProcess --vscode-window-config=vscode:d720a6ce-fc59-491c-a368-442cff9e3aa8 --vscode-window-kind=shared-process
rigr             26504   0.0  0.0 407963472    464 s001  R+    8:33AM   0:00.00 grep 5009`

**Comment by RiccardoGrigoletto** [comment]
Update: I force killed the process Code Helper (Renderer) from activity monitor and restarted VS Code. 
The process Code Helper (Renderer) now looks stable and it's not eating any CPU. 

I am not sure, it could a first-run issue that gets solved automatically.

**Comment by Bovey0809** [comment]
Could you please fix the bug as soon as possible, it makes the Vscode unusable.

**Comment by M00N-MAN** [comment]
Hi

On macOS each (re)starting of VSCode triggers this Code Helper (Renderer/GPU) to consume over 80% of CPU with no observable/obvious reason even if vscode is running in the background (literally doesn't show via WindowsServer)

Version: 1.68.1
Commit: 30d9c6cd9483b2cc586687151bcbcd635f373630
Date: 2022-06-14T12:52:13.188Z
Electron: 17.4.7
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Darwin x64 19.6.0

even if ~/.vscode/settings.json has
"files.exclude": {
        "**/.git": true,
        "**/.svn": true,
        "**/.hg": true,
        "**/CVS": true,
        "**/.DS_Store": true,
        "**/tmp": true,
        "**/node_modules": true,
        "**/bower_components": true,
        "**/dist": true
    },
    "files.watcherExclude": {
        "**/.git/objects/**": true,
        "**/.git/subtree-cache/**": true,
        "**/node_modules/**": true,
        "**/tmp/**": true,
        "**/bower_components/**": true,
        "**/dist/**": true
    }

please let me know how can i help you to collect the necessary logs to engage and resolve this annoying behavior.
![image](https://user-images.githubusercontent.com/7201157/177040109-672c09d0-a9db-41fb-8571-a4b8bec16047.png)

**Comment by nero-angela** [comment]
Same issue in here.
<img width="545" alt="image" src="https://user-images.githubusercontent.com/26322627/178440144-29245fa9-9bb3-4c83-b965-baf1d9c00fc1.png">

**Comment by M00N-MAN** [comment]
> I confirmed that the [settings-sync extension](https://marketplace.visualstudio.com/items?itemName=Shan.code-settings-sync&ssr=false#review-details) was causing the problem, and it was resolved by turning off the extension.
> 
> * [High extenstionHost CPU usage when Settings Sync is enabled shanalikhan/code-settings-sync#1361](https://github.com/shanalikhan/code-settings-sync/issues/1361)

this is offtopic, actually. I have no such extension which you mentioned, and the problem still exists on
Version: 1.69.1
Commit: b06ae3b2d2dbfe28bca3134cc6be65935cdfea6a
Date: 2022-07-12T08:21:51.333Z
Electron: 18.3.5
Chromium: 100.0.4896.160
Node.js: 16.13.2
V8: 10.0.139.17-electron.0
OS: Darwin x64 19.6.0

and it doesn't matter if my extensions are turned ON or OFF (i even tried to temporarily rename the extensions directory in ~/.vscode and restart vscode. it didn't help. 

$ ls -1 ~/.vscode/extensions/
bierner.markdown-mermaid-1.14.2
cheshirekow.cmake-format-0.6.11
go2sh.cmake-integration-vscode-0.7.1
ibm.output-colorizer-0.1.2
llvm-vs-code-extensions.vscode-clangd-0.1.21
mhutchie.git-graph-1.30.0
ms-python.python-2022.10.1
ms-python.vscode-pylance-2022.7.30
ms-python.vscode-pylance-2022.7.40
ms-vscode-remote.remote-containers-0.241.3
ms-vscode-remote.remote-ssh-0.84.0
ms-vscode-remote.remote-ssh-edit-0.80.0
ms-vscode.cmake-tools-1.11.26
ms-vscode.cpptools-1.11.2-darwin-x64
ms-vscode.cpptools-1.11.3-darwin-x64
ms-vscode.cpptools-1.11.4-darwin-x64
nickdemayo.vscode-json-editor-0.3.0
twxs.cmake-0.0.17

Hello @deepak1556, i hope you are doing well

Could you please specify which information you need to push this current ticket forward, and how to get this info for you?

**Comment by CodingCanuck** [comment]
@deepak1556 I took a process sample from OS X's activity monitor when this happened right now on my M1 Macbook Pro. Is there any other information you need someone to capture the next time this happens?

Full process sample text (too large for a single GitHub comment): https://controlc.com/e3f959c0c

**Comment by Aggio98** [comment]
This problem is also happening to me I am a beginner and do not know what to do. Can someone help?

**Comment by amecreate** [comment]
macOS Monterey v12.2.1 new installation of vscode. Opened code then closed. Code Helper caused overheating to 90 C until force quit.

**Comment by M00N-MAN** [comment]
> @deepak1556 I took a process sample from OS X's activity monitor when this happened right now on my M1 Macbook Pro. Is there any other information you need someone to capture the next time this happens?
> 
> Full process sample text (too large for a single GitHub comment): https://controlc.com/e3f959c0c

Dear @CodingCanuck, @bpasero and all, I don't think that @deepak1556 has time for this issue according to [his commitment history](https://github.com/microsoft/vscode/commits?author=deepak1556) since April (current issue started in April as well)

Dear @bpasero could you please share how developers are assigned per issue and how progress is tracked?
Dear @deepak1556  could you please share your ETA, your assumptions about the possible reason, and missing info in this thread to resolve/push this issue?

steps to reproduce:
- open few cpp/python projects/folders in code
- edit some files
- killall code via Terminal at one command 
ps waux| grep -i "Visual Studio Code.app"|grep -v grep|awk '{print $2}'|wc -l
      51
ps waux| grep -i "Visual Studio Code.app"|awk '{print $2}'|xargs -I{} kill -9 {}
ps waux| grep -i "Visual Studio Code.app"|grep -v grep|awk '{print $2}'|wc -l
       0
       
- start code
code
![image](https://user-images.githubusercontent.com/7201157/181311274-5b1f7ad8-a211-42bb-9724-f8e8052597db.png)
after 3-5 minutes the load by the renderer is decreasing
![image](https://user-images.githubusercontent.com/7201157/181312112-19f50919-85c6-479c-8c55-07feb184f887.png)

perhaps some changes could be just reverted until their performance stability aren't well designed?

**Comment by M00N-MAN** [comment]
> macOS Monterey v12.2.1 new installation of vscode from https://code.visualstudio.com/Download. Opened code then closed. Code Helper caused overheating to 90 C until force quit. Edit: installation from Homebrew does not have overheating issue so far.

Dear @amecreate i've just tried vscode from brew.

sudo mv /Applications/Visual\ Studio\ Code.app /Applications/Visual Studio Code.app
sudo mv /usr/local/bin/code /usr/local/bin/code_ms
 brew install visual-studio-code                   
running=brew install visual-studio-code
==> Downloading https://update.code.visualstudio.com/1.69.2/darwin/stable
Already downloaded: /Users/master/Library/Caches/Homebrew/downloads/04932cff610f2130de44f54ea565bb893de9f948a9e43e992abae1a6b3f10fe3--VSCode-darwin.zip
==> Installing Cask visual-studio-code
==> Moving App 'Visual Studio Code.app' to '/Applications/Visual Studio Code.app'
==> Linking Binary 'code' to '/usr/local/bin/code'
🍺  visual-studio-code was successfully installed!

And it didn't help:
After I started the **code** tens of Renderer processes continued consuming CPU for 3-5 minutes.

**Comment by amecreate** [comment]
> 

My workflow:

1. Open Terminal
2. In Terminal, navigate to the directory using cd command
3. Type ```code .``` to open vscode in the desired directory

I noticed that double clicking to open vscode will make it go haywire, at least in my case

**Comment by M00N-MAN** [comment]
> > > > macOS Monterey v12.2.1 new installation of vscode from https://code.visualstudio.com/Download. Opened code then closed. Code Helper caused overheating to 90 C until force quit. Edit: installation from Homebrew does not have overheating issue so far.
> > 
> > 
> > Dear @amecreate i've just tried vscode from brew.
> > sudo mv /Applications/Visual\ Studio\ Code.app /Applications/Visual\ Studio\ Code_MS.app sudo mv /usr/local/bin/code /usr/local/bin/code_ms brew install visual-studio-code running=brew install visual-studio-code ==> Downloading https://update.code.visualstudio.com/1.69.2/darwin/stable Already downloaded: /Users/master/Library/Caches/Homebrew/downloads/04932cff610f2130de44f54ea565bb893de9f948a9e43e992abae1a6b3f10fe3--VSCode-darwin.zip ==> Installing Cask visual-studio-code ==> Moving App 'Visual Studio Code.app' to '/Applications/Visual Studio Code.app' ==> Linking Binary 'code' to '/usr/local/bin/code' 🍺 visual-studio-code was successfully installed!
> > And **it didn't help**: as I already showed above tens of Code Helper Renderer processes continued consuming the CPU for 3-5 minutes just straight after I started the **code**
> 
> My workflow:
> 
> 1. Open Terminal
> 2. In Terminal, navigate to the directory using cd command
> 3. Type `code .` to open vscode in the desired directory
> 
> I noticed that double-clicking to open vscode will make it go haywire, at least in my case

In my case, I'm starting code always from the terminal but not as `code .` but just `code `because I have at least two projects always to work with in parallel: some cpp lib and python project which depends on it. So It is not my option to start code only in one place.

@amecreate could you please share ls -1 ~/.vscode/extensions  to compare with [my list](https://github.com/microsoft/vscode/issues/148226#issuecomment-1193374273)?

**Comment by deepak1556** [comment]
@CodingCanuck thanks for the process sample, the sample currently lacks VSCode version information. Can you provide the output of the menu `Code > About Visual Studio Code`. Thanks!

@M00N-MAN can you provide the process sample of the affected ones showing high cpu usage ? In https://github.com/microsoft/vscode/issues/148226#issuecomment-1197145765 I see the main process launched by executable `/Applications/Visual Studio Code.app/Contents/MacOS/Electron` also showing a high CPU usage. Also include this process when collecting the sample.

Also, can users in this thread check if the issue happens with our [exploration build](https://az764295.vo.msecnd.net/exploration/63d652f7e9193cad178f06be877b8e8f7c63bfc2/VSCode-darwin-universal.zip) that is built with a newer version of the runtime to rule out any runtime related issues.

**Comment by CodingCanuck** [comment]
> @CodingCanuck thanks for the process sample, the sample currently lacks VSCode version information. Can you provide the output of the menu `Code > About Visual Studio Code`. Thanks!

```
Version: 1.69.2 (Universal)
Commit: 3b889b090b5ad5793f524b5d1d39fda662b96a2a
Date: 2022-07-18T16:12:57.074Z
Electron: 18.3.5
Chromium: 100.0.4896.160
Node.js: 16.13.2
V8: 10.0.139.17-electron.0
OS: Darwin arm64 21.5.0
```

> 
> Also, can users in this thread check if the issue happens with our [exploration build](https://az764295.vo.msecnd.net/exploration/63d652f7e9193cad178f06be877b8e8f7c63bfc2/VSCode-darwin-universal.zip) that is built with a newer version of the runtime to rule out any runtime related issues.

**Comment by 7freaks-otte** [comment]
I don't know if this is related, but I was experiencing the same issue for months now until I just had the message "The TypeScript language service died unexpectedly 5 times in the last 5 Minutes." without even having a TypeScript project open.

So I searched and found that issue: https://github.com/microsoft/vscode/issues/88163

@mjbvz suggests to install the extension [ms-vscode.vscode-typescript-next](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next) in order to let VSCode use the latest TypeScript internally (does not affect workspace settings).

And voilà after restarting VSCode the Code Helper process spiked for 5 seconds and then dropped to normal. Hopefully this lasts and might help someone else.

**Comment by theJasonFan** [comment]
FWIW I am also experiencing the issue of `Code Helper (Renderer)` active using CPU when VS Code has been quit. Unfortunately I am having trouble reproducing it.

Version:
```
Version: 1.70.0 (Universal)
Commit: da76f93349a72022ca4670c1b84860304616aaa2
Date: 2022-08-04T04:38:55.829Z
Electron: 18.3.5
Chromium: 100.0.4896.160
Node.js: 16.13.2
V8: 10.0.139.17-electron.0
OS: Darwin arm64 21.6.0
```

Extensions:
```
$ ls -1 ~/.vscode/extensions/
rust-lang.rust-analyzer-0.3.1148-darwin-arm64
```

**Comment by SunshowerC** [comment]
> I confirmed that the [settings-sync extension](https://marketplace.visualstudio.com/items?itemName=Shan.code-settings-sync&ssr=false#review-details) was causing the problem, and it was resolved by turning off the extension.
> 
> * [High extenstionHost CPU usage when Settings Sync is enabled shanalikhan/code-settings-sync#1361](https://github.com/shanalikhan/code-settings-sync/issues/1361)

same problem here. 
In my case, disabled code-settings-sync extension solved the problem.

**Comment by M00N-MAN** [comment]
> > I confirmed that the [settings-sync extension](https://marketplace.visualstudio.com/items?itemName=Shan.code-settings-sync&ssr=false#review-details) was causing the problem, and it was resolved by turning off the extension.
> > 
> > * [High extenstionHost CPU usage when Settings Sync is enabled shanalikhan/code-settings-sync#1361](https://github.com/shanalikhan/code-settings-sync/issues/1361)
> 
> same problem here. In my case, the disabled code-settings-sync extension solved the problem.

fortunately or unfortunately it is off by default on my side. so it doesn't seem to be the root cause or trigger of it

**Comment by zachhardesty7** [comment]
I haven't seen anyone post about this elsewhere, so here's a block of text about what happened to me and what fixed it. I've seen the renderer process eat up a pretty constantly high CPU, getting worse the longer VSCode is open until it eventually freezes the entire program indefinitely (let it sit 30+ minutes).

For me, VSCode would only start getting worse **while the "Problems" panel was open**. It would get bad enough that trying to type a word in the editor would freeze for a second or two after every letter. Switching to another panel like "Terminal" and back would entirely freeze VSCode for multiple seconds, but even just sitting there, not touching anything with the window hidden, would cause high CPU when "Problems" was open.

I'm running the latest insiders build on a multi-thousand file TypeScript project (plus dependencies) with a `tsc --watch` running from a VSCode task with a problem matcher set up. I profiled extensions, `tsserver`, etc so many times and removed some things causing slowdowns. This improved editor performance, sometimes notably, but I continued my descent into the madness of a laggy editor.

I profiled the window from "JavaScript profiler" in the VSCode devtools today and triggered a handful of lag spikes. I saw 1 specific function (in `workbench.desktop.main.js`, named `_`) that seemed to be taking up to 5 seconds to run in the flame graph. In minified form, I could only tell it was something regex related. Hit it with a breakpoint and figured out it was **the filter bar in the "Problems" panel. Removing the filter text seems to have fixed my issues.** I don't think the filter I had was crazy, just filtering out a few files, but seems it checks as often as the problems panel updates, which can be every keystroke. Not sure if it's been getting slower in VSCode's side, but we've added a bunch of new ESLint rules and other things that show up in the "Problems" panel to our project recently. For the record, the filter text looked like this: `!**/FileName1.scss,!**/FileName2.js,!**/angular.js,!**/FileName3.tsx[],!**/?/settings.json,!**/build/**,!**/?/workspaceSettings.json,!**/FileName5.jsp`

It's probably not the exact same issue as OP, but hopefully this helps someone!

**Comment by foxly-it** [comment]
Same Issue here with my MacBook Air M2 :-/

**Comment by brownsugar** [comment]
Same problem for months...

It doesn't ALWAYS drain the battery but OFTEN.
Sometimes I restarts VSCode the problem will be resolved but after a few hours or another day, the problem could happened again.

<img width="828" alt="image" src="https://user-images.githubusercontent.com/2935980/185850131-9c688766-133d-4787-b35b-d9fe8b1883d1.png">

<img width="586" alt="image" src="https://user-images.githubusercontent.com/2935980/185849365-67db9971-e40b-408c-b635-714d2aff4c36.png">

```
Version: 1.70.2 (Universal)
Commit: e4503b30fc78200f846c62cf8091b76ff5547662
Date: 2022-08-16T05:36:37.829Z
Electron: 18.3.5
Chromium: 100.0.4896.160
Node.js: 16.13.2
V8: 10.0.139.17-electron.0
OS: Darwin x64 21.6.0
```

**Comment by Detnator-Neo** [comment]
I'm having a slightly different but I suspect related problem. Fully loaded MacBook Pro 16" M1 Max.   I'm coding away and every few seconds VSCode stops responding to clicks and keystrokes (but no beachball).  Then a few seconds later all clicks and keystrokes I did while it was unresponsive then happen in order.

Running Activity Monitor next to it so I can see at a glance what might be happening, every time this happens I see `Code Helper (Renderer)` process eating 96+% of CPU.  Then a few seconds later, that process in Activity Monitor drops to near zero, at the same moment VSCode becomes responsive again.

Unlike others above, it's happening while I'm coding, not with VSCode quit, and it's only happening momentarily, although I think more frequently and for a little longer each time, I think.

If this should be a separate issue, let me know and I'll post it as such.

**Comment by AmyTheCute** [comment]
I also was having this issue, Removing the app from the applications folder did not help. I ran -rm commands on folders I found it to install from google searches, that improved the heat a lot but my laptop was still hotter and lasting 6h instead of the usual 12-15h (browsing) so I finally found that force quitting Code Helper from Visual studio code in activity monitor helped me fix the issue, I wonder if it'll pop back

**Comment by JasonJayKim** [comment]
> I don't know if this is related, but I was experiencing the same issue for months now until I just had the message "The TypeScript language service died unexpectedly 5 times in the last 5 Minutes." without even having a TypeScript project open.
> 
> So I searched and found that issue: #88163
> 
> @mjbvz suggests to install the extension [ms-vscode.vscode-typescript-next](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next) in order to let VSCode use the latest TypeScript internally (does not affect workspace settings).
> 
> And voilà after restarting VSCode the Code Helper process spiked for 5 seconds and then dropped to normal. Hopefully this lasts and might help someone else.

This helps me!!!

**Comment by M00N-MAN** [comment]
> @CodingCanuck thanks for the process sample, the sample currently lacks VSCode version information. Can you provide the output of the menu `Code > About Visual Studio Code`. Thanks!
> 
> @M00N-MAN can you provide the process sample of the affected ones showing high cpu usage ? In [#148226 (comment)](https://github.com/microsoft/vscode/issues/148226#issuecomment-1197145765) I see the main process launched by executable `/Applications/Visual Studio Code.app/Contents/MacOS/Electron` also showing a high CPU usage. Also include this process when collecting the sample.
> 
> Also, can users in this thread check if the issue happens with our [exploration build](https://az764295.vo.msecnd.net/exploration/63d652f7e9193cad178f06be877b8e8f7c63bfc2/VSCode-darwin-universal.zip) that is built with a newer version of the runtime to rule out any runtime related issues.
Hi @deepak1556
Please let me know if this is enough or not
[Sample of Code Helper2.txt](https://github.com/microsoft/vscode/files/9693283/Sample.of.Code.Helper2.txt)
[Spindump2.txt](https://github.com/microsoft/vscode/files/9693284/Spindump2.txt)
[Spindump.txt](https://github.com/microsoft/vscode/files/9693285/Spindump.txt)
[Sample of Code Helper.txt](https://github.com/microsoft/vscode/files/9693286/Sample.of.Code.Helper.txt)

**Comment by M00N-MAN** [comment]
> I haven't seen anyone post about this elsewhere, so here's a block of text about what happened to me and what fixed it. I've seen the renderer process eat up a pretty constantly high CPU, getting worse the longer VSCode is open until it eventually freezes the entire program indefinitely (let it sit 30+ minutes).
> 
> For me, VSCode would only start getting worse **while the "Problems" panel was open**. It would get bad enough that trying to type a word in the editor would freeze for a second or two after every letter. Switching to another panel like "Terminal" and back would entirely freeze VSCode for multiple seconds, but even just sitting there, not touching anything with the window hidden, would cause high CPU when "Problems" was open.
> 
> I'm running the latest insiders build on a multi-thousand file TypeScript project (plus dependencies) with a `tsc --watch` running from a VSCode task with a problem matcher set up. I profiled extensions, `tsserver`, etc so many times and removed some things causing slowdowns. This improved editor performance, sometimes notably, but I continued my descent into the madness of a laggy editor.
> 
> I profiled the window from "JavaScript profiler" in the VSCode devtools today and triggered a handful of lag spikes. I saw 1 specific function (in `workbench.desktop.main.js`, named `_`) that seemed to be taking up to 5 seconds to run in the flame graph. In minified form, I could only tell it was something regex related. Hit it with a breakpoint and figured out it was **the filter bar in the "Problems" panel. Removing the filter text seems to have fixed my issues.** I don't think the filter I had was crazy, just filtering out a few files, but seems it checks as often as the problems panel updates, which can be every keystroke. Not sure if it's been getting slower in VSCode's side, but we've added a bunch of new ESLint rules and other things that show up in the "Problems" panel to our project recently. For the record, the filter text looked like this: `!**/FileName1.scss,!**/FileName2.js,!**/angular.js,!**/FileName3.tsx[],!**/?/settings.json,!**/build/**,!**/?/workspaceSettings.json,!**/FileName5.jsp`
> 
> It's probably not the exact same issue as OP, but hopefully this helps someone!
Hi @zachhardesty7

your story looks pretty good, but I can't confirm it is close to the root cause because in my case I have only:
- a few opened vscode windows with folders or files (some are in not saved state),
- only c++ or python
- "problems" windows have something from clangd or pylance
- all vscode windows with all Problems nor other tabs/windows have empty filters
- I don't have any ts code/project/files. :(

**Comment by M00N-MAN** [comment]
> > I don't know if this is related, but I was experiencing the same issue for months now until I just had the message "The TypeScript language service died unexpectedly 5 times in the last 5 Minutes." without even having a TypeScript project open.
> > So I searched and found that issue: #88163
> > @mjbvz suggests to install the extension [ms-vscode.vscode-typescript-next](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next) in order to let VSCode use the latest TypeScript internally (does not affect workspace settings).
> > And voilà after restarting VSCode the Code Helper process spiked for 5 seconds and then dropped to normal. Hopefully this lasts and might help someone else.
> 
> This helps me!!! (M1 pro macOS Monterey v12.4) In my case, disabled [ms-vscode.vscode-typescript-next](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next) extension solved the problem.

Hi @JasonJayKim
so obviously this is not vscode issue and offtopic in this thread
![image](https://user-images.githubusercontent.com/7201157/193472906-415d397c-b06d-4713-b69e-4ccbd0130572.png)

**Comment by M00N-MAN** [comment]
Closed vscode...
Code helper still in progress
![image](https://user-images.githubusercontent.com/7201157/193473128-34fa3372-bd97-4eb5-90a1-2e7f2fb82594.png)
[Spindump4.txt](https://github.com/microsoft/vscode/files/9693346/Spindump4.txt)
[Spindump3.txt](https://github.com/microsoft/vscode/files/9693347/Spindump3.txt)
[Sample of Code Helper4.txt](https://github.com/microsoft/vscode/files/9693348/Sample.of.Code.Helper4.txt)
[Sample of Code Helper3.txt](https://github.com/microsoft/vscode/files/9693349/Sample.of.Code.Helper3.txt)
it has something about pylance in it's command line, but this is different from what I had recently. turning off pylance and restarting to see the result

**Comment by ajmas** [comment]
I too am running into an issue where I am running into high CPU, though the process explorer seems to have a granularity issue, meaning it is hard to identify the real culprit:

<img width="1169" alt="image" src="https://user-images.githubusercontent.com/682269/193708170-475a904f-c9e9-487b-8846-b7f78a30e49f.png">

Here we see "shared-process" at 66% CPU, yet nothing below show any CPU usage. Using `top -u` on the CLI confirms the 66% CPU usage of "Code Helper".

Copying the output of the process explorer, to get processes:

```
shared-process
  ptyHost
  fileWatcher
window (OpeningsService.ts — scarlet-staff-server)

extensionHost

/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node --max-old-space-size=3072 /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js --serverMode partialSemantic --useInferredProjectPerProjectRoot --disableAutomaticTypingAcquisition --cancellationPipeName /var/folders/sj/j454ybpx3tbbngp006kmm5f00000gn/T/vscode-typescript501/577e1bf040b11523300c/tscancellation-1a8a4434617b7e04b648.tmp* --globalPlugins @vsintellicode/typescript-intellicode-plugin,ms-vsintellicode-typescript --pluginProbeLocations /Users/ajmas/.vscode/extensions/visualstudioexptteam.vscodeintellicode-1.2.28,/Users/ajmas/.vscode/extensions/visualstudioexptteam.vscodeintellicode-1.2.28 --locale en --noGetErrOnBackgroundUpdate --validateDefaultNpmLocation --useNodeIpc
/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node --max-old-space-size=3072 /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/tsserver.js --useInferredProjectPerProjectRoot --enableTelemetry --cancellationPipeName /var/folders/sj/j454ybpx3tbbngp006kmm5f00000gn/T/vscode-typescript501/577e1bf040b11523300c/tscancellation-463b1cb45c1a5ecc682b.tmp* --globalPlugins @vsintellicode/typescript-intellicode-plugin,ms-vsintellicode-typescript --pluginProbeLocations /Users/ajmas/.vscode/extensions/visualstudioexptteam.vscodeintellicode-1.2.28,/Users/ajmas/.vscode/extensions/visualstudioexptteam.vscodeintellicode-1.2.28 --locale en --noGetErrOnBackgroundUpdate --validateDefaultNpmLocation --useNodeIpc
  /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/typingsInstaller.js --globalTypingsCacheLocation /Users/ajmas/Library/Caches/typescript/4.8 --enableTelemetry --typesMapLocation /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/node_modules/typescript/lib/typesMap.json --validateDefaultNpmLocation
/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node /Users/ajmas/.vscode/extensions/shinnn.stylelint-0.51.0/server.js --stdio --clientProcessId=59829
/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node /Users/ajmas/.vscode/extensions/formulahendry.auto-rename-tag-0.1.10/packages/server/dist/serverMain.js --node-ipc --clientProcessId=59829
/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node /Users/ajmas/.vscode/extensions/kisstkondoros.vscode-gutter-preview-0.30.0/dist/server.js --node-ipc --clientProcessId=59829
/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node /Users/ajmas/.vscode/extensions/visualstudioexptteam.intellicode-api-usage-examples-0.2.4/dist/server/server.js --node-ipc --clientProcessId=59829
/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node /Users/ajmas/.vscode/extensions/dbaeumer.vscode-eslint-2.2.6/server/out/eslintServer.js --node-ipc --clientProcessId=59829
/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node /Users/ajmas/.vscode/extensions/vue.volar-0.40.13-darwin-x64/server --node-ipc --clientProcessId=59829
/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node /Users/ajmas/.vscode/extensions/vue.volar-0.40.13-darwin-x64/server --node-ipc --clientProcessId=59829
/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/json-language-features/server/dist/node/jsonServerMain --node-ipc --clientProcessId=59829
```

Environment:

  - Intel based MacBook Pro (16-inch, 2019) 
  - VS Code 1.71.2
  - macOS 12.15.1

**Comment by M00N-MAN** [comment]
Hi @deepak1556,

Could you please share which tool you use for analysis of performance issues with the given spindump/samples of vscode in this thread? (maybe some link to the guide about it)

When we can receive any info/update/workaround for the current issue from your side?

How else can we help you in resolving this issue?

with best regards

**Comment by M00N-MAN** [comment]
I just didn't get it, can you reproduce it or not
![image](https://user-images.githubusercontent.com/7201157/196026734-efd8e45e-c706-4069-a4e1-6bd0c8a35576.png)
![image](https://user-images.githubusercontent.com/7201157/196026800-91356acf-52b4-48c5-93dc-af52cb205fc6.png)

**Comment by M00N-MAN** [comment]
![image](https://user-images.githubusercontent.com/7201157/196034200-6500b771-7e4f-4598-8e5d-ddd34a7fe4c8.png)

**Comment by maxoppelt** [comment]
The issue also occurs with

- VS Code Version:
```bash
code --version
1.72.2
d045a5eda657f4d7b676dedbfa7aab8207f8a075
x64
electron 19.0.17
```
- OS Version:
```bash
sw_vers
ProductName:	macOS
ProductVersion:	12.6.1
BuildVersion:	21G217
```

Steps to Reproduce:
```
ps aux <PID>
<USER>  <PID> 100,0  2,8 71937192 471344   ??  R    10:51am   9:25.75 /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node /Users/<USER>/.vscode/extensions/ms-python.vscode-pylance-2022.10.40/dist/server.bundle.js --canc
```

Probably Related:
- *Issue:* https://github.com/microsoft/vscode/issues/148226
- *Issue*: https://github.com/microsoft/vscode/issues/161494
- *Issue*: https://github.com/microsoft/vscode/issues/165016
- *Issue*: https://github.com/microsoft/vscode/issues/160098
- *Issue*: https://github.com/microsoft/vscode/issues/147707
- *Plugin Pylance:*: https://github.com/microsoft/pylance-release

Already in settings:
```json
"files.exclude": {
    "**/__pycache__": true,
    "**/.pytest_cache": true,
    "**/.git": true,
    "**/.DS_Store": true,
    "**/tmp": true,
    "**/node_modules": true,
    "**/bower_components": true,
    "**/dist": true
},

"files.watcherExclude": {
    "**/.git/objects/**": true,
    "**/.git/subtree-cache/**": true,
    "**/node_modules/**": true,
    "**/tmp/**": true,
    "**/bower_components/**": true,
    "**/dist/**": true
},
```

**Comment by AurevoirXavier** [comment]
> https://github.com/mhutchie/vscode-git-graph/issues/712

Same issue.

Once I open the git-graph plugin, vscode will freeze.

**Comment by M00N-MAN** [comment]
> > [mhutchie/vscode-git-graph#712](https://github.com/mhutchie/vscode-git-graph/issues/712)
> 
> Same issue.
> 
> Once I open the git-graph plugin, vscode will freeze.

Hi, I don't confirm it.

Rather I say the opposite:
mhutchie.git-graph isn't impacting that topic at all from my side.

Turning off the git graph extension and restarting the vscode: the same problem with CPU load in the beginning
Turning on the git graph extension doesn't make any additional load, nor before nor after the restart of the vscode.

Thanks for the idea, nice try)

**Comment by M00N-MAN** [comment]
> The issue also occurs with
> 
> * VS Code Version:
> 
> ```shell
> code --version
> 1.72.2
> d045a5eda657f4d7b676dedbfa7aab8207f8a075
> x64
> electron 19.0.17
> ```
> 
> * OS Version:
> 
> ```shell
> sw_vers
> ProductName:	macOS
> ProductVersion:	12.6.1
> BuildVersion:	21G217
> ```
> 
> Steps to Reproduce:
> 
> ```
> ps aux <PID>
> <USER>  <PID> 100,0  2,8 71937192 471344   ??  R    10:51am   9:25.75 /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node /Users/<USER>/.vscode/extensions/ms-python.vscode-pylance-2022.10.40/dist/server.bundle.js --canc
> ```
> 
> Probably Related:
> 
> * _Issue:_ [VSCode’s renderer helper process using 100% of CPU #148226](https://github.com/microsoft/vscode/issues/148226)
> * _Issue_: [Code Helper uses 180% CPU #161494](https://github.com/microsoft/vscode/issues/161494)
> * _Issue_: [VSCode becomes laggy after using it for some time (macOS Ventura) #165016](https://github.com/microsoft/vscode/issues/165016)
> * _Issue_: [Code Helper process take 100% of my CPU #160098](https://github.com/microsoft/vscode/issues/160098)
> * _Issue_: [Bug: Battery drain even when app off! #147707](https://github.com/microsoft/vscode/issues/147707)
> * _Plugin Pylance:_: https://github.com/microsoft/pylance-release
> 
> Already in settings:
> 
> ```json
> "files.exclude": {
>     "**/__pycache__": true,
>     "**/.pytest_cache": true,
>     "**/.git": true,
>     "**/.DS_Store": true,
>     "**/tmp": true
> },
> 
> "files.watcherExclude": {
>     "**/.git/objects/**": true,
>     "**/.git/subtree-cache/**": true,
>     "**/node_modules/**": true,
>     "**/tmp/**": true,
> },
> ```

Hi, I've tried to turn off any pylance-related extensions. A clean start with turned-off extension (pylance) doesn't make any difference to the renderer: its instances still are consuming a lot of CPU during the first 5-10 minutes.

Thanks

**Comment by M00N-MAN** [comment]
Hi @hediet, Could you please share your opinion about why this topic doesn't have any valuable updates for so long time?

**Comment by M00N-MAN** [comment]
Hi @deepak1556, Could you please share your opinion about why this topic doesn't have any valuable updates for so long time?

How could attendees of this topic help you to push it forward?

**Comment by dmitryshelomanov** [comment]
+ 1 
<img width="381" alt="image" src="https://user-images.githubusercontent.com/19505559/209482712-0816bd9a-6f56-4ade-a13f-04059f6a6ae0.png">

**Comment by ozgurrgul** [comment]
I think the record is mine.

<img width="977" alt="image" src="https://user-images.githubusercontent.com/20097288/210012658-919cd907-b273-49bc-971b-04174435782b.png">

**Comment by ozgurrgul** [comment]
@aka-vm Thanks for suggestion. I didn't see improvement when I turn off extension. (Since as I mentioned the only extension I use is `prettier`)

**Comment by aka-vm** [comment]
@ozgurrgul 
hmmm! Try using 1. by combing 2.
Today I was having this problem, it was fixed after I did that.

I hope you find a solution soon. Else switching to an older version of **vscode** and other things could also work.

**Comment by HenryC-3** [comment]
> > [mhutchie/vscode-git-graph#712](https://github.com/mhutchie/vscode-git-graph/issues/712)
> > [gitkraken/vscode-gitlens#2389](https://github.com/gitkraken/vscode-gitlens/issues/2389)
> 
> Same issue.
> 
> Once I open the git-graph plugin, vscode will freeze.

same here

**Comment by Panayiotou-P** [comment]
Same problem also. Very sad that I don't see any solution yet

**Comment by Down10** [comment]
I am pretty disappointed that this CPU bug has gone unfixed, and meanwhile Atom has sunsetted. I always preferred Atom as my MacOS source editor, not because it was better than VS Code, but because it didn't spike my CPU to 100% every time I opened it.

**Comment by SolomidHero** [comment]
same problem :(
killed process that was ~100% cpu, and another started in few seconds with already ~220% cpu

**Comment by kirksw** [comment]
I also have this issue, using 440%+ of my cpu

**Comment by ajmas** [comment]
> I also have this issue, using 440%+ of my cpu

I am no a dev on the project, but providing more info, such as OS, VSCode version and any extensions could be useful. By itself the comment doesn't add much for the devs to base their investigation on, if they decide to do so.

**Comment by LucasArruda** [comment]
I also have this process

`/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/MacOS/Code Helper (Plugin) --type=utility --utility-sub-type=node.mojom.NodeService --lang=en-US --service-sandbox-type=none --inspect-port=0 --user-data-dir=/Users/lucasarruda/Library/Application Support/Code --standard-schemes=vscode-webview,vscode-file --secure-schemes=vscode-webview,vscode-file --bypasscsp-schemes --cors-schemes=vscode-webview,vscode-file --fetch-schemes=vscode-webview,vscode-file --service-worker-schemes=vscode-webview --streaming-schemes --shared-files --field-trial-handle=1718379636,r,11794770712285918391,468314003805687192,131072 --disable-features=CalculateNativeWinOcclusion,SpareRendererForSitePerProcess`

And it takes like 50~80% CPU basically forever.

I wonder if that's like VSCode itself.

**Comment by sangit0** [comment]
Same issue with for m2 mac pro it takes 70-80% CPU ,

/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/MacOS/Code Helper (Plugin)

**Comment by LucasArruda** [comment]
I found out that it looks like that most of this seems to come from CoPilot? Or maybe the main process that supports extensions, and upon killing the process, Github CoPilot is one of the processes that gets killed (it goes away and the comes back in the lower VSCode bar).

So, maybe it's all due to CoPilot processing our codebases?

Do you have it also, @sangit0 ?

**Comment by LucasArruda** [comment]
I've seen some supporting evidence of this issue occurring before (though they state, in some tickets, it's been fixed)

https://github.com/orgs/community/discussions/8861

https://github.com/community/community/discussions/7889

https://github.com/microsoft/vscode/issues/147188

I don't want to disable it, though, because I use it a lot. It just can be a bummer if I keep the editor open when idle and it keeps with that 60% constant CPU usage.

**Comment by tamuratak** [comment]
Known causes so far on macOS Ventura (M1 or M2)

- Setting Sync extension: https://marketplace.visualstudio.com/items?itemName=Shan.code-settings-sync
  - Disable it. See #147707
- CMake Tools: https://marketplace.visualstudio.com/items?itemName=ms-vscode.cmake-tools
  - Disable it. See #145387. Reported at upstream microsoft/vscode-cmake-tools/issues/2967
- ESLint extension: https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint
  - Disable it. Go to #171691
- (❗THIS THREAD❗) Renderer process hits 100% of CPU
  - One of causes is Git Lens, Git Graph, and other extensions which heavily uses canvas elements on WebView. Disable them.
  - (❗new❗) Minimap might be the cause. Disable Minimap. See https://github.com/microsoft/vscode/issues/169775#issuecomment-1422919512

**Comment by tamuratak** [comment]
Pro tips for trouble shooting:

- Use the built-in [Process Explorer](https://code.visualstudio.com/updates/v1_23#_see-the-running-visual-studio-code-processes) instead of Activity Monitor by Apple. Process names are more helpful.
- If `extension-host` process hits 100% of CPU, one of extensions is the cause. Find it with [bisect](https://code.visualstudio.com/blogs/2021/02/16/extension-bisect).
- If `main`, `shared-process`, or `window (...)` process hits 100% of CPU, it might be a issue on VS Code's side. Open a new issue. VS Code members will determine whether it is a duplicate.

**Comment by sergei3000** [comment]
Same issue here.
This is the process:
```
ps aux | grep 35428
username   35428 200.0  0.8 72010812 137084   ??  R     3:49PM  36:56.23 /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node /Users/username/.vscode/extensions/ms-python.vscode-pylance-2023.2.20/dist/server.bundle.js --cancellationReceive=file:787703fe46c8fcfe746a022668193a1bdb8ca7d483 --node-ipc --clientProcessId=35305
```

**Comment by tamuratak** [comment]
@sergei3000 

No, it isn't the same issue. You should report to https://github.com/microsoft/pylance-release/issues

**Comment by ripperdoc** [comment]
There are clearly many different ways VS Code could get into a situation where it runs high on CPU. Is there a way to find which Extension is using the CPU? Turning them on and off is not an easy way to find the issue if it's intermittent, it would be much better to be able to print which extensions are taking what part of the Code Helper process load.

**Comment by sangit0** [comment]
> I found out that it looks like that most of this seems to come from CoPilot? Or maybe the main process that supports extensions, and upon killing the process, Github CoPilot is one of the processes that gets killed (it goes away and the comes back in the lower VSCode bar).
> 
> So, maybe it's all due to CoPilot processing our codebases?
> 
> Do you have it also, @sangit0 ?

For me it was the code spell checker and setting sync plugin, both were taking so much CPU. I disabled them and got significant improvement over the CPU usages.

**Comment by tamuratak** [comment]
@sangit0 @LucasArruda 

Issues you are seeing are not related to this thread. Please report to the repositories of the extensions causing your issues.

**Comment by zubricks** [comment]
This is still happening as of today.. VSCode was closed, and CodeHelper was using 100% of my CPU. What's going on here?

**Comment by ahough306** [comment]
New M2 airbook, clean vscode install w/ no extensions. Same issue after install, opened to see if it was working but remained hot after. Used to edit two text files. Saw it was using around 75% cpu consistently.

**Comment by sergei3000** [comment]
I just quit vscode on my mac and am still having two Code Helper processes each of which is running at 200% CPU

**Comment by sergei3000** [comment]
This is ridiculous. vscode should not work like this even if some extension is failing. When I quit vscode all of its processes should stop, obviously.

**Comment by tamuratak** [comment]
@sergei3000 It could happen. For example, see #175713.

Anyway, you had better open a new issue. It is not sure if your issue is the same one as this thread.

**Comment by jung-han** [comment]
<img width="346" alt="image" src="https://user-images.githubusercontent.com/35371660/224173027-8699f653-982b-49aa-966b-4f87cea3bc02.png">
Same here.
<img width="229" alt="image" src="https://user-images.githubusercontent.com/35371660/224173179-e0523cc2-1202-4d6b-9943-7604f6d81326.png">

**Comment by tamuratak** [comment]
@jung-han 

Your issue is not different one. It is about `Code Helper (Plugin)`. This thread is about `Code Helper (Renderer)`.

You had better open a new issue.

**Comment by ginobean** [comment]
I don't know if this is related, but I had some issues with VSCode using up high loads. Then, it eventually occurred to me that I was working off an nfs share drive (network drive). Moved my work over to a local drive and haven't encountered significant high load issues since.. 

As a side note, it might be interesting to see how much VSCode performance is impacted when working off an nfs or samba network drive.. There should be ways to maintain good performance even in these cases..

**Comment by atre** [comment]
Disabling setting sync worked for me

**Comment by tamuratak** [comment]
@ginobean 

Your issue seems different. You had better open a new issue, which would help the improvement of VS Code.

>  it eventually occurred to me that I was working off an nfs share drive (network drive).

**Comment by ginobean** [comment]
That's a good point. I think I mentioned the problem I encountered because it's possible that some other people, who were having 
load/performance issues re: VSCode, might be encountering the same issue. Personally, I had forgotten that I was working off an nfs share drive and was a bit mystified as to why some operations seemed to take so long, with high loads. It was only until later, that I realized it was likely due to working off an nfs share..

**Comment by tamuratak** [comment]
Now, VS Code Insiders 1.78 has updated Electron to v22.4.3. It would be worth checking to see if this update fixes the issue.

- https://code.visualstudio.com/insiders/
- #177600

**Comment by tamuratak** [comment]
> Version: 1.78.0-insider solved this problem.  (Electron: 22.4.3)
> Looking forward to the release of the stable version.

_Originally posted by @jqtmviyu in https://github.com/microsoft/vscode/issues/178934#issuecomment-1500138160_

**Comment by mnmkng** [comment]
Happened to me today, clean install on M1 Pro macOS: 13.3.1 (22E261). Opened one Python file, installed Python extension. an hour later I realized my battery is dying fast, so I quit VSCode, but the battery kept draining fast and I found 2 Code Helper (renderer) processes eating 250% CPU.

**Comment by M00N-MAN** [comment]
Hi @deepak1556 @LucasArruda 
Do you have any plans/ideas about current report/thread?

**Comment by M00N-MAN** [comment]
> Happened to me today, clean install on M1 Pro macOS: 13.3.1 (22E261). Opened one Python file, installed Python extension. an hour later I realized my battery is dying fast, so I quit VSCode, but the battery kept draining fast and I found 2 Code Helper (renderer) processes eating 250% CPU.

Hi

could you please share your list of extensions active for that time/such case?

how about to check 
for mem cosumers
watch -n1 -d "ps -wweo pid,pcpu,pmem,rss,vsz,args -m |grep -v 'watch'"
or for cpu consumers
watch -n1 -d "ps -wweo pid,pcpu,pmem,rss,vsz,args -r |grep -v 'watch'"

**Comment by M00N-MAN** [comment]
> Now, VS Code Insiders 1.78 has updated Electron to v22.4.3. It would be worth checking to see if this update fixes the issue.
> 
> * https://code.visualstudio.com/insiders/
> * [chore: update to Electron 22 #177600](https://github.com/microsoft/vscode/pull/177600)

currently i don't see this issue
on macos 13.3 mac m1 pro max
with
Version: 1.77.3 (Universal)
Commit: 704ed70d4fd1c6bd6342c436f1ede30d1cff4710
Date: 2023-04-12T09:19:37.325Z
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Darwin arm64 22.4.0
Sandboxed: No

but latest version (different arch?) has same topic issue as i've shown along time ago above
![image](https://user-images.githubusercontent.com/7201157/235482807-e08f1e99-7945-4372-a62b-f30c7d139913.png)

**Comment by kyb3r** [comment]
I got a new MacBook Pro and when I installed vscode and started doing dev work I was losing battery fast and my laptop was warm. I was kinda confused because I thought this laptop was supposed to be efficient. Took a look in my task manager today and vscode is there hogging 600% cpu even when it was closed. Damn.

**Comment by ArchieAtkinson** [comment]
I'm also getting this on a MacBook Pro 2017 running MacOS 13.4 (22F66) and VSCode 1.78.2 (Universal) with a single extension (yzhang.markdown-all-in-one). This issue happens even when VSCode is not open.

**Comment by LucasArruda** [comment]
@M00N-MAN my problem, apparently, went away with recent updates.

I had one last remaining process that was taking too much CPU and I was able to determine it was not vscode, not github copilot, but a ruby server (solargraph).

Upon the last vscode updates and by removing solargraph, I now have sane usage and a cool M1 (which was getting hot – which is not so easy to pull on an M1/M1 Pro).

**Comment by andefred** [comment]
Any updates on this? The only pattern I can see from the people posting and my own issue is that We are (mostly) using the same older version of MacOs namely some version of Monterey
![image](https://github.com/microsoft/vscode/assets/957302/f97e77f8-ee69-49c9-bac3-8702f95c8bd0)

Here's a screenshot of the process what when it's running/spinning out of control, the TS intellisense stops working or becomes incredibly slow. Could this be related to some tsconfig issue? I'm in an NX monorepo with not that many apps/libraries, but this makes VS code unusable... :( 
![image](https://github.com/microsoft/vscode/assets/957302/6285cc88-7ae0-4008-8176-354e952815f2)

**Comment by tamuratak** [comment]
@andefred 

Your issue is different from this thread. Your issue is about `Code Helper (Plugin)`. This thread is about `Code Helper (Renderer)`.

You had better open a new issue.

See also https://github.com/microsoft/vscode/issues/148226#issuecomment-1426483095

**Comment by M00N-MAN** [comment]
> @M00N-MAN my problem, apparently, went away with recent updates.
> 
> I had one last remaining process that was taking too much CPU and I was able to determine it was not vscode, not github copilot, but a ruby server (solargraph).
> 
> Upon the last vscode updates and by removing solargraph, I now have sane usage and a cool M1 (which was getting hot – which is not so easy to pull on an M1/M1 Pro).

happy for you, but i didn't and don't have/use any solargraph or ruby.
cheers

**Comment by M00N-MAN** [comment]
> @sergei3000
> 
> No, it isn't the same issue. You should report to https://github.com/microsoft/pylance-release/issues

@tamuratak i dunno why you/github decided that i can't mention you:)
but Code Render (originally in current case https://github.com/microsoft/vscode/issues/148226) has 100% issue no matter if i have pylance or not. i've checked that. So in my subjective point of view you can't just subdivide someones report only because it contains the pylance because this pylance extension can't be executed separately/outside of Code Helper which is part of complain/report in current case. 

Could you please at least how to start vscode with full logging to get more info about it? Thanks for 0 answers btw for such long period after all info had been shared as you asked above

**Comment by KamenRiderKuuga** [comment]
Having the same issue on my M2 Macbook Air with macOS Ventura 13.3, and the vscode version is 1.80.1, stable build.

Reboot my computer solve the problem, but it's kind of disturbing.

**Comment by chrishenderson07** [comment]
I have the same Problem.
Even debugging the Vscode with no extension installed, the high power consumption persists.

We need a solution for that.

This will be a big blow from Microsoft against AppleX_X?

**Comment by gregfenton** [comment]
Not sure if this is the exact same but symptoms seems similar.

I am on a MacBook Pro (Intel).  Our project has a directory named `output` that is added to the .gitignore.  We run scripts inside the project directory that generate thousands of files into the `output` directory.  For the most part, everything seems to be just fine.

... UNTIL ... we run `rm -rf output/*` in a terminal (in VSCode or outside of VSCode).  Then VSCode seems to go to 100%, `top` shows that it is CodeHelper running hard, and VSCode becomes quite unresponsive (not fully unresponsive...but many things don't seem to respond to keystrokes or mouse clicks properly).  Also, the Source Control tab shows a clock icon that does not go away.

Moving that `output` directory to be outside of the VSCode project seems to have reduced (eliminated??) the problem.

**Comment by growthcode** [comment]
For me, the issue turned out to be files that needed to be excluded.  I only recently switched to VScode and it was searching my SQL files.  

Searching "asdf" ended up crashing VS code once 32 gigs of RAM became occupied by the search editor.

**Comment by sebastien-roch** [comment]
I launched VS code from the terminal and disabled the `sync` feature:
```
code --disable-extensions --verbose --sync off
```
Then I opened the command palette (command shift p on mac), typed "Show synced data" and restored a UI state from before I had the issue. After this, I relaunched VS code normally (not from Terminal) and the issue was gone, no more `renderer` process eating up my CPU and making VS Code totally unusable.

I hope this helps someone else.

**Comment by chrishenderson07** [comment]
So What's Solution? do you install a old version to confirm

> I confirm issue is still present in latest version of VS Code: Version: 1.82.2 (Universal) Commit: [abd2f3d](https://github.com/microsoft/vscode/commit/abd2f3db4bdb28f9e95536dfa84d8479f1eb312d) Date: 2023-09-14T05:59:47.790Z Electron: 25.8.1 ElectronBuildId: 23779380 Chromium: 114.0.5735.289 Node.js: 18.15.0 V8: 11.4.183.29-electron.0 OS: Darwin x64 22.6.0

**Comment by Fy-** [comment]
this `ext install ms-vscode.vscode-typescript-next` fixed it for me.

**Comment by letanloc1998** [comment]
I have the same issue with Code Helper. It used a lot of CPU on Mac M1. It's better after I kill it.

**Comment by ericblade** [comment]
Fresh VSCode install, only extension is GitHub Pull Requests.  Upon starting VSC, the Code Helper (Renderer) pegs at exactly 283% and stays there until the Mac is rebooted.

**Comment by mnlmaier** [comment]
+1, same problem here — makes using the Application rather impossible for me :/

**Comment by caschb** [comment]
+1 on an M3 Pro machine.

**Comment by letanloc1998** [comment]
+1 on a Mac M1

**Comment by dfcastellanos** [comment]
I had high CPU usage, which my case was caused by the extension Pylance. Downgrading did not fix it but disabling it did. See [https://github.com/microsoft/pylance-release/issues/4885](https://github.com/microsoft/pylance-release/issues/4885) and [https://github.com/microsoft/pylance-release/issues/201](https://github.com/microsoft/pylance-release/issues/201).

**Comment by bwmills** [comment]
Any updates on this issue?

**Comment by joulei** [comment]
I'm having the same issues on a Macbook Pro 2023 M2 24 GB ram, running Sonoma 14.2.1 (23C71).
VSCode singlehandedly maxes out at least 2 CPUs and brings the laptop to unusable temperature. 
I'm only running Rust Analyzer and a custom theme.

**Comment by NeonWizard** [comment]
Likewise, just started today. No change in settings or extensions.

**Comment by chrisdel101** [comment]
This is still happening to me runaway code helper processes that keep respawning after being killed

**Comment by 1scriptz** [comment]
Here I am about to take my mac mini m1 into Apple and this thread appears. 

Running MacOS 14.4.1 and just started happening to me 4-5 days ago.

**Comment by ajmas** [comment]
I’d suspect in many cases it is an extension acting up, but unfortunately given the current task manager UI, it is hard knowing which one. 

I’m going to check the task manager to see if that is still the case (not at my computer right now).

**Comment by ericblade** [comment]
> I’d suspect in many cases it is an extension acting up, but unfortunately given the current task manager UI, it is hard knowing which one.
> 
> I’m going to check the task manager to see if that is still the case (not at my computer right now).

I don't boot to MacOS much, but mine only had github pull requests installed, and it wasn't active.

**Comment by 1scriptz** [comment]
> I’d suspect in many cases it is an extension acting up, but unfortunately given the current task manager UI, it is hard knowing which one.
> 
> I’m going to check the task manager to see if that is still the case (not at my computer right now).

This was a fresh install of MacOS/VSCode. Nothing was changed, no extensions, nothing. 

What concerned me the most was the machine potentially running for 8hrs while I slept with this process at 200% cpu usage/baking my mac.

**Comment by xitex** [comment]
for me this problem always happened after asyncio run with some urls to visit with httpx.client.... it's time to change this editor. microsoft as always....

**Comment by joulei** [comment]
> I'm having the same issues on a Macbook Pro 2023 M2 w/ 24 GB RAM, running Sonoma 14.2.1 (23C71). VSCode singlehandedly maxes out at least 3 cores and brings the laptop to unusable temperature. I'm only running Rust Analyzer and a custom theme, no python extensions or settings sync.

After a couple of months, I found two solutions for my case: 
- https://www.gnu.org/software/emacs/
- https://zed.dev/

**Comment by bebimoj** [comment]
What's happening to VS Code lately? It seems to become more and more useless with every update. Unfortunately, updates now occur almost daily, making me wonder if the developers know what they're doing. It now spawns ten different Clang processes, each consuming 100% CPU. This seems to be related to Intellisense, as the entire editor freezes whenever I start typing. It must be the TypeScript language server causing the issue , i'm so sick of  typescript language server! Invested so much time in VSCode now have to seek another editor.

**Comment by gregfenton** [comment]
> What's happening to VS Code lately? [...] updates now occur almost daily [...]

How did you come to that conclusion?  Are you on some experimental install or something because I am seeing updates about [once per month as documented here](https://code.visualstudio.com/updates/v1_91)

**Comment by M00N-MAN** [comment]
> I had high CPU usage, which my case was caused by the extension Pylance. Downgrading did not fix it but disabling it did. See [microsoft/pylance-release#4885](https://github.com/microsoft/pylance-release/issues/4885) and [microsoft/pylance-release#201](https://github.com/microsoft/pylance-release/issues/201).

could you please read the content of this thread?
there are few reports that pylance wasn't installed.

Thank you in advance :)

**Comment by xrstf** [comment]
v1.91.1 on Kubuntu 24.04 -- still happening. Right after starting one process just fries a CPU core, sometimes for a few minutes, sometimes for hours, sometimes for days.

**Comment by SuichiM** [comment]
> Update: I force killed the process Code Helper (Renderer) from activity monitor and restarted VS Code. The process Code Helper (Renderer) now looks stable and it's not eating any CPU.
> 
> I am not sure, it could be a first-run issue that gets solved automatically.

in this case, m3 pro, macOS Sonoma 14.5, with no sync-plugin. it was solved following this (by now)!

**Comment by gregfenton** [comment]
> in this case, m3 pro, macOS Sonoma 14.5, with no sync-plugin. it was solved following this (by now)!

This doesn't solve the problem....it just makes it go away for a while.

**Comment by SuichiM** [comment]
So, No final solution yet?

> > in this case, m3 pro, macOS Sonoma 14.5, with no sync-plugin. it was solved following this (by now)!
> 
> This doesn't solve the problem....it just makes it go away for a while.

**Comment by jefferson-artium** [comment]
It just happened to me. I'm using Linux Pop!_OS. I disabled the built-in settings sync, and the problem stopped.

Version: 1.92.0
Commit: b1c0a14de1414fcdaa400695b4db1c0799bc3124
Date: 2024-07-31T23:26:45.634Z
Electron: 30.1.2
ElectronBuildId: 9870757
Chromium: 124.0.6367.243
Node.js: 20.14.0
V8: 12.4.254.20-electron.0
OS: Linux x64 6.9.3-76060903-generic

**Comment by ashemedai** [comment]
Happened to me, fresh install, no extensions, no projects open. Just the VS Code main program with Welcome tab.

macOS Sequoia 15.0.1 on M3 MacBook Pro
VS Code 1.94.2 (Universal)

After quitting VS Code itself to resolve the issue, the following processes still stuck around:
Code Helper (Renderer) 292.1% CPU, CPU Time 34:17:35.98
Code Helper (Plugin) 197.0% CPU, CPU Time 23:02:03.48

Are there any guides on how to get started with troubleshooting/debugging this issue?

**Comment by SupLano** [comment]
I just started having the same issues, similar setup

> Same issue with my Macbook Air M1, 2020 Sequoia: VScode becomes unacceptable due to this. The MB get too hot.

**Comment by dragnag** [comment]
MacBook Pro M2, MacOS Ventura: same issue after recent auto-update (and also occurring in the vanilla insiders VS Code installation), solved by disabling "Workbench > Tree: Render Indent Guides" in the app settings (set to `none`). Hope this helps someone.
